tests/infer/test_abstract_infer.py

```diff
@@ -81,6 +88,19 @@ def test_posterior_predictive_svi_auto_diag_normal_guide():
     assert_close(marginal_return_vals.mean, torch.ones(5) * 700, rtol=0.05)
 
 
+def test_posterior_predictive_svi_one_hot():
+    pseudocounts = torch.ones(3) * 0.1
+    true_probs = torch.tensor([0.15, 0.6, 0.25])
+    classes = dist.OneHotCategorical(true_probs).sample((1000,))
+    opt = optim.Adam(dict(lr=0.1))
+    loss = Trace_ELBO()
+    guide = AutoDiagonalNormal(one_hot_model)
+    svi_run = SVI(one_hot_model, guide, opt, loss, num_steps=1000, num_samples=100).run(pseudocounts, classes=classes)
```

Comment:
This test still seems to fail - I think you might want to increase the `num_samples` to `1000`.

Reply:
OK, I will try again 😄 . It's a bit weird it kept failing, since I ran it 10 times locally without any issue